### PR TITLE
Initial commit for Bindings Quickstart for Javascript, go, Python and…

### PR DESCRIPTION
**Pull request by nirmash**
… C# (http and SDK)

Signed-off-by: Nir Mashkowski <nimashkowski@microsoft.com>

# Description

Created a Quickstart for Bindings to add to Dapr.io docs

## Issue reference

n/a

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] The quickstart code compiles correctly
* [x] You've tested new builds of the quickstart if you changed quickstart code
* [x] You've updated the quickstart's README if necessary
* [no will need help creating a test for this Quickstart ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

File: bindings/csharp/http/input/input.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+var builder = WebApplication.CreateBuilder(args);
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment()) {app.UseDeveloperExceptionPage();}
+
+// Dapr Kafka input binding
+
+app.MapPost("/sample-topic", (Order requestData) => {
```
Comment (amulyavarote) [inline bindings/csharp/http/input/input.cs:30]: Can we keep it as "/orders" (like we have in pubsub examples)?

File: bindings/csharp/sdk/output/output.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+using System;
+using Dapr.Client;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+var bindingName = "sample-topic";
```
Comment (amulyavarote) [inline bindings/csharp/sdk/output/output.cs:18]: Can we keep it as "orders"? Sample-Topic is very generic

File: bindings/go/http/input.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Golang - Kafka HTTP input binding: ", string(data))
+}
+
+func main() {
+	daprPort := ":6002"
+	bindingName := "/sample-topic"
```
Comment (amulyavarote) [inline bindings/go/http/input.go:37]: Same as above

File: bindings/go/http/output.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+func main() {
+	daprHost := "http://localhost"
+	if value, ok := os.LookupEnv("DAPR_HOST"); ok {
+		daprHost = value
+	}
+	daprHttpPort := "6061"
+	if value, ok := os.LookupEnv("DAPR_HTTP_PORT"); ok {
+		daprHttpPort = value
+	}
+	bindingName := "sample-topic"
```
Comment (amulyavarote) [inline bindings/go/http/output.go:37]: Same as above

File: bindings/go/sdk/input.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/dapr/go-sdk/service/common"
+	daprd "github.com/dapr/go-sdk/service/http"
+)
+
+func main() {
+	bindingName := "/sample-topic"
```
Comment (amulyavarote) [inline bindings/go/sdk/input.go:28]: same as above

File: bindings/go/sdk/output.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	"fmt"
+	"strconv"
+	"time"
+
+	dapr "github.com/dapr/go-sdk/client"
+)
+
+//code
+func main() {
+	bindingName := "sample-topic"
```
Comment (amulyavarote) [inline bindings/go/sdk/output.go:29]: Can we keep it as "orders" sample-topic is very generic?

File: bindings/javascript/http/output.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// dapr run --app-id javascript-output-binding-http node output.js --components-path ../../components
+
+import axios from "axios";
+
+const DAPR_HOST = process.env.DAPR_HOST || "http://localhost";
+const DAPR_HTTP_PORT = process.env.DAPR_HTTP_PORT || "3500";
+const KAFKA_TOPIC = process.env.DAPR_KAFKA_TOPIC || "sample-topic";
```
Comment (amulyavarote) [inline bindings/javascript/http/output.js:19]: Lets try to keep the variable names consistent across all the programming languages. (Keep one: kafka_topic or binding_topic). And also keep it as "orders"

File: bindings/javascript/sdk/input.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import { DaprServer } from "dapr-client";
+
+const daprHost = "127.0.0.1"; 
+const daprPort = "5051"; 
+const serverHost = "127.0.0.1";
+const serverPort = "3500";
+
+async function start() {
+  const server = new DaprServer(serverHost, serverPort, daprHost, daprPort);;
+  const bindingName = "sample-topic";
```
Comment (amulyavarote) [inline bindings/javascript/sdk/input.js:23]: Same as above

File: bindings/python/http/input.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+from flask import Flask, request, jsonify
+import logging
+
+app = Flask(__name__)
+
+logger = logging.getLogger('werkzeug')
+
+# Dapr input binding
+@app.route('/sample-topic', methods=['POST'])
```
Comment (amulyavarote) [inline bindings/python/http/input.py:23]: Same as above!

File: bindings/python/http/output.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# dapr run --app-id python-output-binding-http --app-port 4002 --dapr-http-port 4001 python3 output.py --components-path ../../components
+
+import time
+import requests
+
+dapr_port = 4001
+binding_name = "sample-topic"
```
Comment (amulyavarote) [inline bindings/python/http/output.py:18]: Same as above. Lets keep it as "orders"

File: bindings/python/sdk/input.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# dapr run --app-id python-input-binding-sdk --app-protocol grpc --app-port 50051 --components-path ../../components python3 input.py
+# pip3 install dapr dapr-ext-grpc
+
+from dapr.ext.grpc import App, BindingRequest
+
+app = App()
+bindingName = "sample-topic"
```
Comment (amulyavarote) [inline bindings/python/sdk/input.py:18]: Same as above

File: bindings/python/sdk/output.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+# limitations under the License.
+# 
+
+import json
+import time
+
+from dapr.clients import DaprClient
+
+with DaprClient() as d:
+    bindingName = "sample-topic"
```
Comment (amulyavarote) [inline bindings/python/sdk/output.py:20]: Same as above

**Comment by amulyavarote** [comment]
lets try to keep the variable names, method names and print statements consistent across all the languages.

File: bindings/python/sdk/output.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    operation = "create"
+    n = 0
+    while n < 10:
+        n += 1
+        req_data = {
+            'orderId': n
+        }
+        print ('Python - Kafka SDK output binding: orderId: ' + str(n),flush=True)
```
Comment (amulyavarote) [inline bindings/python/sdk/output.py:28]: Lets keep the print statements same across

File: bindings/python/http/input.py [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+def process_binding():
+    event_orderid = request.data.decode('utf-8')
+    logger.info('Python - Kafka HTTP input binding: ' + event_orderid)
```
Comment (amulyavarote) [inline bindings/python/http/input.py:26]: No need to have the language specified in the print statements.

File: bindings/javascript/sdk/output.js [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+
+async function main() {
+  const client = new DaprClient(daprHost, httpPort);
+  const bindingName = "sample-topic";
+  for(var i = 1; i <= 10; i++) {
+    const order = {orderId:  i};
+    
+    // Publish a Kafka event using an output binding
+    await client.binding.send(bindingName, "create", order);
+    console.log("Javascript - Kafka SDK output binding: " + JSON.stringify(order));
```
Comment (amulyavarote) [inline bindings/javascript/sdk/output.js:28]: No need to have the language specified in the print statements.

File: bindings/javascript/sdk/input.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+  const response = await server.binding.receive(bindingName, async (data) => console.log(`Javascript - Kafka SDK input binding: ${JSON.stringify(data)}`));
```
Comment (amulyavarote) [inline bindings/javascript/sdk/input.js:24]: No need to have the language specified in the print statements.

File: bindings/javascript/http/output.js [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+async function main() {
+  for(var i = 1; i <= 10; i++) {
+    const order = { data :
+        {orderId: i},
+        operation: "create"
+    };
+    // Publish a Kafka event using an output binding
+    await axios.post(`${DAPR_HOST}:${DAPR_HTTP_PORT}/v1.0/bindings/${KAFKA_TOPIC}`, order)
+      .then(function (response) {
+        console.log("Javascript - Kafka HTTP output binding: " + response.config.data);
```
Comment (amulyavarote) [inline bindings/javascript/http/output.js:30]: No need to have the language specified in the print statements.

File: bindings/javascript/http/input.js [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+import bodyParser from "body-parser";
+import "isomorphic-fetch";
+
+const app = express();
+app.use(bodyParser.json());
+
+const port = 3500;
+
+app.post('/sample-topic', (req, res) => {
+    console.log("Javascript - Kafka HTTP input binding: " + JSON.stringify(req.body));
```
Comment (amulyavarote) [inline bindings/javascript/http/input.js:25]: No need to have the language specified in the print statements.

File: bindings/go/sdk/output.go [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+		client, err := dapr.NewClient()
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+		ctx := context.Background()
+		//Using Dapr SDK to invoke output binding
+		in := &dapr.InvokeBindingRequest{Name: bindingName, Operation: bindingOperation, Data: []byte(order)}
+		err = client.InvokeOutputBinding(ctx, in)
+		fmt.Println("Golang - Kafka SDK output binding: ", order)
```
Comment (amulyavarote) [inline bindings/go/sdk/output.go:43]: No need to have the language specified in the print statements.

File: bindings/go/sdk/input.go [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+		log.Fatalf("error adding binding handler: %v", err)
+	}
+
+	if err := s.Start(); err != nil && err != http.ErrServerClosed {
+		log.Fatalf("error listenning: %v", err)
+	}
+}
+
+func runHandler(ctx context.Context, in *common.BindingEvent) (out []byte, err error) {
+	fmt.Println("Golang - Kafka SDK input binding: ", string(in.Data))
```
Comment (amulyavarote) [inline bindings/go/sdk/input.go:42]: No need to have the language specified in the print statements.

File: bindings/go/http/output.go [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+			log.Fatal(err.Error())
+			os.Exit(1)
+		}
+
+		// Publish a Kafka event
+		if _, err = client.Do(req); err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Println("Golang - Kafka HTTP output binding: ", order)
```
Comment (amulyavarote) [inline bindings/go/http/output.go:53]: No need to have the language specified in the print statements.

File: bindings/go/http/input.go [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+
+	"github.com/gorilla/mux"
+)
+
+func processBinding(w http.ResponseWriter, r *http.Request) {
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Golang - Kafka HTTP input binding: ", string(data))
```
Comment (amulyavarote) [inline bindings/go/http/input.go:32]: No need to have the language specified in the print statements.

File: bindings/csharp/sdk/output/output.cs [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+
+for (int i = 1; i <= 10; i++) {
+    var order = new Order(i);
+    using var client = new DaprClientBuilder().Build();
+
+    // Publish a Kafka message using output binding
+    await client.InvokeBindingAsync(bindingName, "create", order);
+    Console.WriteLine("C# - Kafka SDK output binding: " + order);
```
Comment (amulyavarote) [inline bindings/csharp/sdk/output/output.cs:26]: No need to have the language specified in the print statements.

File: bindings/csharp/http/output/output.cs [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+httpClient.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+
+for (int i = 1; i <= 10; i++) {
+    var order = new Order(i);
+    var daprData = new DaprData(order, "create");
+    var orderJson = JsonSerializer.Serialize<DaprData>(daprData);
+    var content = new StringContent(orderJson, Encoding.UTF8, "application/json");
+    // Publish an event/message using Dapr PubSub via HTTP Post
+    var response = httpClient.PostAsync($"{baseURL}/v1.0/bindings/{bindingName}", content);
+    Console.WriteLine("C# - Kafka HTTP output binding: " + orderJson);
```
Comment (amulyavarote) [inline bindings/csharp/http/output/output.cs:35]: No need to have the language specified in the print statements.

File: bindings/csharp/http/input/input.cs [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+    Console.WriteLine("C# - Kafka HTTP input binding: { \"orderId\": " + requestData.OrderId + "}");
```
Comment (amulyavarote) [inline bindings/csharp/http/input/input.cs:31]: No need to have the language specified in the print statements.

**Review by amulyavarote — CHANGES_REQUESTED** [review]
Please make the changes to the print statements, variable names, method names and binding topic.

**Comment by amulyavarote** [comment]
@nirmash Can you rebase this PR with master and push the changes for the builds to be successful?

**Comment by nirmash** [comment]
I am in the process of rewriting the Quickstart, will take a bit of time and I will rebase and resubmit. I am not that experienced with pull requests but perhaps we can just close \ cancel this one?

**Pull request closed without merging**